pkg/kube/snapshot/snapshot.go

```diff
@@ -84,32 +82,30 @@ type Snapshotter interface {
 	//
 	// 'name' is the name of the VolumeSnapshot that will be cloned.
 	// 'namespace' is the namespace of the VolumeSnapshot that will be cloned.
-	// 'cloneName' is name of the clone.
-	// 'cloneNamespace' is the namespace where the clone will be created.
 	// 'waitForReady' will make the function blocks until the clone's status is ready to use.
-	// 'labels' is the labels to set on the created VSC
-	Clone(ctx context.Context, name, namespace, cloneName, cloneNamespace string, waitForReady bool, labels map[string]string) error
+	// 'snapshotMeta' has metadata of the Volumesnaphot resource that is going to get created.
+	// 'contentMeta' has metadata of the VolumesnapshotContent content resource that is going to get created.
+	Clone(ctx context.Context, name, namespace string, waitForReady bool, snapshotMeta, contentMeta ObjectMeta) error
 	// GetSource will return the CSI source that backs the volume snapshot.
 	//
 	// 'snapshotName' is the name of the Volumesnapshot.
 	// 'namespace' is the namespace of the Volumesnapshot.
 	GetSource(ctx context.Context, snapshotName, namespace string) (*Source, error)
-	// CreateFromSource will create a 'Volumesnapshot' and 'VolumesnaphotContent' pair for the underlying snapshot source.
+	// CreateFromSource will create a 'Volumesnapshot' and 'VolumesnapshotContent' pair for the underlying snapshot source.
 	//
 	// 'source' contains information about CSI snapshot.
-	// 'snapshotName' is the name of the snapshot that will be created.
-	// 'namespace' is the namespace of the snapshot.
 	// 'waitForReady' blocks the caller until snapshot is ready to use or context is cancelled.
-	// 'labels' is the labels to set on the created VSC
-	CreateFromSource(ctx context.Context, source *Source, snapshotName, namespace string, waitForReady bool, labels map[string]string) error
-	// CreateContentFromSource will create a 'VolumesnaphotContent' for the underlying snapshot source.
+	// 'annotations' is the annotation to set on the created VS and VSC
+	// 'snapshotMeta' has metadata of the Volumesnapshot resource that is going to get created.
+	// 'contentMeta' has metadata of the VolumesnapshotContent resource that is going to get created.
```

Comment:
```suggestion
	// 'snapshotMeta' has metadata of the VolumeSnapshot resource that is going to get created.
	// 'contentMeta' has metadata of the VolumeSnapshotContent resource that is going to get created.
```
Lets fix this for all the occurrences